builtin/logical/mysql/path_config_connection.go

```diff
@@ -58,6 +71,12 @@ func (b *backend) pathConnectionWrite(
 	return nil, nil
 }
 
+type connectionConfig struct {
+	// Deprecate "value" in coming releases
+	ConnectionString   string `json:"connection_string" json:"value"`
```

Comment:
The alternative would be to add a struct member called e.g. ConnectionURL, then have the code look for that and only use ConnectionString if ConnectionURL is empty.